### PR TITLE
Remove docstrigs if __ONNX_NO_DOC_STRINGS

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: Remove some docstrings if __ONNX_NO_DOC_STRINGS is defined. I'm also looking into other ways to make operator registration smaller/less costly.

### CODE REVIEW

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 1]
```diff
@@ -208,43 +223,44 @@ Performs element-wise binary {name} on 8 bit data types (with Numpy-style broadc
...
     ONNX_NAMESPACE::ReplaceAll(doc, "{additionalDocumentation}", additionalDocumentation);
     schema.SetDoc(doc);
-    schema.Input(0, "A", "First operand.", "T");
+#endif
+    schema.Input(0, "A", "First operand."_docstring, "T");
     schema.Input(
         1,
         "A_scale",
-        "Input A's scale. It's a scalar, which means a per-tensor/layer quantization.",
+        "Input A's scale. It's a scalar, which means a per-tensor/layer quantization."_docstring,
```
Comment (linnealovespie) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:231]: So you're adding the option to reduce the documentation to just the input schema. What's the _docstring operator/how does that work?

Comment (tiagoshibata) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:231]: It's in line 161: https://github.com/microsoft/onnxruntime/pull/4494/files#diff-2e8813db1626a69ab79902bd2f63e556R161-R169

```
constexpr const char* operator "" _docstring(const char* string, size_t) {
#ifdef __ONNX_NO_DOC_STRINGS
    (void)string;
    return "";
#else
    return string;
#endif
}
```

Comment (tiagoshibata) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:231]: `(void)string;` is a no-op, it's there just to prevent a warning on unused parameters.

Comment (linnealovespie) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:231]: Oh OK, the using statement was throwing me off but makes sense.

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 2]
```diff
@@ -158,13 +158,23 @@ void convTransposeWithDynamicPadsShapeInference(InferenceContext& ctx) {
     return;
   }
 }
+
+constexpr const char* operator "" _docstring(const char* string, size_t) {
```
Comment (smk2007) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:162]: super cool usage of user defined literals

**Comment by martinb35** [comment]
what is the size difference with this change?

**Comment by pranavsharma** [comment]
FYI: For the Edge browser builds, we disable contrib ops altogether by using the disable_contrib_ops build flag.

**Comment by tiagoshibata** [comment]
> what is the size difference with this change?

A few KB sadly :( My goal was more to start the practice of using this and start applying this elsewhere so it adds up to more savings in the future.

But I'm looking into more savings in operator registration.

**Pull request merged**